plugins/scheme-dice/main.go

```diff
@@ -17,36 +17,31 @@ import (
 
 var altNameID = asn1.ObjectIdentifier{2, 5, 29, 17}
 
-type EvidenceExtractor struct {
+type Scheme struct {
 }
 
-func (ee EvidenceExtractor) GetName() string {
-	return "DICE"
+func (s Scheme) GetName() string {
+	return common.AttestationFormat_DICE.String()
 }
 
-// Init is a no-op for this plugin.
-func (ee EvidenceExtractor) Init(params common.EvidenceExtractorParams) error {
-	return nil
-}
-
-// Close is a no-op for this plugin.
-func (ee EvidenceExtractor) Close() error {
-	return nil
+func (s Scheme) GetFormat() common.AttestationFormat {
+	return common.AttestationFormat_DICE
 }
 
-func (ee EvidenceExtractor) GetTrustAnchorID(token []byte) (common.TrustAnchorID, error) {
-	return common.TrustAnchorID{Type: common.TaTypeCert}, nil
+func (s Scheme) GetTrustAnchorID(token *common.AttestationToken) (string, error) {
+	return "dice://", nil
 }
 
-func (ee EvidenceExtractor) ExtractEvidence(token []byte, trustAnchor []byte) (map[string]interface{}, error) {
+//func (s Scheme) ExtractEvidence(token []byte, trustAnchor []byte) (map[string]interface{}, error) {
+func (s Scheme) ExtractEvidence(token *common.AttestationToken, trustAnchor string) (*common.ExtractedEvidence, error) {
```

Comment:
Would be good to refer to a DICE Standard in comments that we have followed when ExtractEvidence, having known that so many variations of DICE in the market ?

Reply:
This a hypothetical DICE example (vaguely based on RIOT, as it was defined at the time it was created). It does not really intend to follow any particular standard.